server/src/main/scala/io/delta/standalone/internal/DeltaSharedTableLoader.scala

```diff
@@ -111,10 +121,7 @@ class DeltaSharedTable(
       limitHint: Option[Int]): (Long, Seq[model.SingleAction]) = withClassLoader {
     // TODO Support `limitHint`
     val snapshot = deltaLog.snapshot
-    if (snapshot.version < 0) {
-      throw new IllegalStateException(s"The table ${tableConfig.getName} " +
-        s"doesn't exist on the file system or is not a Delta table")
-    }
+    validateDeltaTable()
```

Comment:
nit: Could you pass the snapshot to `validateDeltaTable`. It's better to use the same snapshot in a single request.